src/components/tools/drawing-tool/drawing-toolbar.tsx

```diff
@@ -90,71 +106,51 @@ export const ToolbarView: React.FC<IProps> = (
-                              onSetSelectedButton={handleSetSelectedButton} />
-            {
-              currentStamp &&
-              <div
-                  className={"flyout-top-button " + modalButtonClasses("stamp")}
-                  style={{height: 30}} title="Coin Stamp"
-                onClick={() => handleSetSelectedButton("stamp")}>
-                {
-                  stamps.length > 1 &&
-                  <div className="flyout-toggle" onClick={handleStampListButton}>
-                    ▶
-                  </div>
-                }
-                <img src={currentStamp.url} />
-              </div>
-            }
-            <ClassIconButton content={drawingContent} modalButton={{ disabled: !drawingContent.hasSelectedObjects }}
-                  title="Delete" iconClass="bin" onClick={handleDeleteButton} />
+            <SvgToolModeButton {...modalButtonProps("select", {})}
+                                title="Select" onSetSelectedButton={handleSetSelectedButton} />
+            <SvgToolModeButton {...modalButtonProps("line", { fill: drawingContent.stroke })}
+                                title="Freehand" onSetSelectedButton={handleSetSelectedButton} />
+            <SvgToolModeButton {...modalButtonProps("vector")}
+                                title="Line" onSetSelectedButton={handleSetSelectedButton} />
+            <SvgToolModeButton {...modalButtonProps("rectangle")} title="Rectangle"
+                                onSetSelectedButton={handleSetSelectedButton} />
+            <SvgToolModeButton {...modalButtonProps("ellipse")} title="Ellipse"
+                                onSetSelectedButton={handleSetSelectedButton} />
+            {currentStamp &&
+              <StampModeButton stamp={currentStamp} stampCount={stampCount} title="Stamp"
+                                selected={drawingContent.isSelectedButton("stamp")}
+                                onClick={handleStampsButtonClick} onTouchHold={handleStampsButtonTouchHold} />}
+            <StrokeColorButton settings={drawingContent.toolbarSettings}
+                  onClick={() => handleToggleShowStrokeColorPalette()} />
+            <FillColorButton settings={drawingContent.toolbarSettings}
+                  onClick={() => handleToggleShowFillColorPalette()} />
+            <DeleteButton disabled={!drawingContent.hasSelectedObjects} onClick={handleDeleteButton} />
           </div>
-          {showSettings
-            ? <DrawingSettingsView
-                drawingContent={drawingContent}
-                colors={colors}
-                onStrokeChange={handleStrokeChange}
-                onFillChange={handleFillChange}
-                onStrokeDashArrayChange={handleStrokeDashArrayChange}
-                onStrokeWidthChange={handleStrokeWidthChange} />
+          {paletteState.showStroke
+            ? <StrokeColorPalette selectedColor={drawingContent.stroke} onSelectColor={handleStrokeColorChange} />
+            : null}
```

Comment:
A few of these could elimate the `null` and be condensed to just:
`{isFoo && <ReactComponent />}`